src/tabnineChatWidget/ChatViewProvider.ts

```diff
@@ -24,17 +24,26 @@ export default class ChatViewProvider implements WebviewViewProvider {
 
   private extensionPath: string;
 
+  private isChatInitiated: boolean = false;
```

Comment:
can't you check on the callbacks array if you are alreay registered? keeping a value might cause more pain then help (or is it something like register with an identifier that will make sure that multiple calls are handled once)

Reply:
we don't have two-way communication from the extension to the app. once we have it, we will change this behaviour.